python/paddle/vision/ops.py

```diff
@@ -1351,3 +1353,152 @@ def __init__(self,
         if activation_layer is not None:
             layers.append(activation_layer())
         super().__init__(*layers)
+
+
+def nms(boxes, threshold):
+    r"""
+    This operator implements non-maximum suppression. Non-maximum suppression (NMS)
+    is used to select one box out of many overlapping boxes in object detection. 
+    Boxes with IoU > threshold will be considered as overlapping boxes, 
+    just one with highest score can be kept. Here IoU is Intersection Over Union, 
+    which can be computed by:
+
+    ..  math::
+
+        IoU = \frac{intersection\_area(box1, box2)}{union\_area(box1, box2)}
+
+    Args:
+        boxes(Tensor): The input boxes data to be computed, it's a 2D-Tensor with 
+            the shape of [num_boxes, 4] and boxes should be sorted by their 
+            confidence scores. The data type is float32 or float64. 
+            Given as [[x1, y1, x2, y2], …],  (x1, y1) is the top left coordinates, 
+            and (x2, y2) is the bottom right coordinates. 
+            Their relation should be ``0 <= x1 < x2 && 0 <= y1 < y2``.
+        threshold(float32): IoU threshold for determine overlapping boxes.
+
+    Returns:
+        Tensor: 1D-Tensor with the shape of [num_boxes]. Indices of boxes kept by NMS.
+
+    Examples:
+        .. code-block:: python
+        
+            import paddle
+            import numpy as np
+
+            boxes = np.random.rand(4, 4).astype('float32')
+            boxes[:, 2] = boxes[:, 0] + boxes[:, 2]
+            boxes[:, 3] = boxes[:, 1] + boxes[:, 3]
+
+            # [[0.46517828 0.90615124 1.3940141  1.8840896 ]
+            # [0.74385834 0.8236293  1.4048514  1.3868837 ]
+            # [0.39436954 0.18261194 1.3834884  0.38191944]
+            # [0.9617653  0.40089446 1.2982695  1.398673  ]]
+
+            out =  paddle.vision.ops.nms(paddle.to_tensor(boxes), 0.1)
+            # [0, 2, 0, 0])
+    """
+    if _non_static_mode():
+        return _C_ops.nms(boxes, 'iou_threshold', threshold)
+
+    helper = LayerHelper('nms', **locals())
+    out = helper.create_variable_for_type_inference('int64')
+    helper.append_op(
+        type='nms',
+        inputs={'Boxes': boxes},
+        outputs={'KeepBoxesIdxs': out},
+        attrs={'iou_threshold': threshold})
+    return out
+
+
+def batched_nms(boxes, scores, category_idxs, categories, iou_threshold, top_k):
```

Comment:
please also add unit testing for  `Dynamic  to Static `

Reply:
added test_batched_nms_dynamic_to_static function in python/paddle/fluid/tests/unittest/test_batched_nms.py